chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiUpdater.java

```diff
@@ -458,35 +458,35 @@ public void checkForDuplicates() throws SQLException, Exception
-		duplicatesLog.info("# DB_ID\tCreator\tDuplicated Identifier\tReferenceMolecule");
-
-		// Should only be one, but API returns collection.
-		@SuppressWarnings("unchecked")
-		Collection<GKInstance> chebiDBInsts = (Collection<GKInstance>)adaptor.fetchInstanceByAttribute(ReactomeJavaConstants.ReferenceDatabase, ReactomeJavaConstants.name, "=", "ChEBI");
-		GKInstance chebiDBInst = chebiDBInsts.stream().findFirst().orElse(null);
-		
-		AttributeQueryRequest chebiAQR = adaptor.new AttributeQueryRequest(ReactomeJavaConstants.ReferenceMolecule, ReactomeJavaConstants.referenceDatabase, "=", chebiDBInst);
-		
-		while (duplicates.next())
+		try(ResultSet duplicates = adaptor.executeQuery(findDuplicateReferenceMolecules, null))
 		{
-			String identifier = duplicates.getString(1);
-			int numberOfDuplicates = duplicates.getInt(2);
-			this.duplicatesSB.append("\n** ReferenceMolecule with identifier " + identifier + " occurs " + numberOfDuplicates + " times:\n\n");
-			String operator = identifier == null ? "IS NULL" : "=";
+			duplicatesLog.info("# DB_ID\tCreator\tDuplicated Identifier\tReferenceMolecule");
+	
+			// Should only be one, but API returns collection.
+			@SuppressWarnings("unchecked")
+			Collection<GKInstance> chebiDBInsts = (Collection<GKInstance>)adaptor.fetchInstanceByAttribute(ReactomeJavaConstants.ReferenceDatabase, ReactomeJavaConstants.name, "=", "ChEBI");
+			GKInstance chebiDBInst = chebiDBInsts.stream().findFirst().orElse(null);
 			
-			AttributeQueryRequest identifierAQR = adaptor.new AttributeQueryRequest(ReactomeJavaConstants.ReferenceMolecule, ReactomeJavaConstants.identifier, operator, identifier);
+			AttributeQueryRequest chebiAQR = adaptor.new AttributeQueryRequest(ReactomeJavaConstants.ReferenceMolecule, ReactomeJavaConstants.referenceDatabase, "=", chebiDBInst);
 			
-			@SuppressWarnings("unchecked")
-			Collection<GKInstance> dupesOfIdentifier = (Collection<GKInstance>) adaptor._fetchInstance(Arrays.asList(chebiAQR, identifierAQR));
-			for (GKInstance duplicate : dupesOfIdentifier)
+			while (duplicates.next())
 			{
-				GKInstance creator = ChebiUpdater.getCreator(duplicate);
-				duplicatesLog.info("{}\t{}\t{}\t{}", duplicate.getDBID(), creator.toString(), identifier, duplicate.toString());
+				String identifier = duplicates.getString(1);
+				int numberOfDuplicates = duplicates.getInt(2);
+				this.duplicatesSB.append("\n** ReferenceMolecule with identifier " + identifier + " occurs " + numberOfDuplicates + " times:\n\n");
+				String operator = identifier == null ? "IS NULL" : "=";
+				
+				AttributeQueryRequest identifierAQR = adaptor.new AttributeQueryRequest(ReactomeJavaConstants.ReferenceMolecule, ReactomeJavaConstants.identifier, operator, identifier);
+				
+				@SuppressWarnings("unchecked")
+				Collection<GKInstance> dupesOfIdentifier = (Collection<GKInstance>) adaptor._fetchInstance(Arrays.asList(chebiAQR, identifierAQR));
+				for (GKInstance duplicate : dupesOfIdentifier)
```

Comment:
Most of the logic here seems to be setting up getting the `dupesofIdentifier` collection.  Could that functionality be its own method that takes in the identifier value from each loop?  The method could handle the construction of the details of building the AttributeQueryRequests.

Reply:
It's only 3 or 4 lines, I'm not sure what the problem here is.

Reply:
I was just getting lost in the details of creating the AQRs.  It's okay to keep it as is if you prefer.  I just thought it would help focus the intent of the logic here to capture the details fetching of the duplicate identifier into a method and then call it.

Reply:
Some comments have been added, hopefully that will make it easier to follow.